coral/src/app/features/topics/details/subscriptions/TopicSubscriptionsTable.tsx

```diff
@@ -202,14 +205,33 @@ export const TopicSubscriptionsTable = ({
   filteredData,
   onDelete,
   onDetails,
+  isLoading,
 }: TopicSubscriptionsTableProps) => {
   const rows = getRows(selectedSubs, filteredData);
   const columns = getColumns(selectedSubs, onDelete, onDetails);
+  const loadingColumns = columns.map((col) => {
+    return {
+      headerName: col.headerName,
+      width: col.width,
+      headerVisible: col.headerInvisible,
+    };
+  });
 
-  return rows.length === 0 ? (
-    <EmptyState title="No subscriptions">
-      No subscription matched your criteria.
-    </EmptyState>
+  if (rows.length === 0) {
+    return (
+      <EmptyState title="No subscriptions">
+        {!isLoading && <>No subscription matched your criteria.</>}
+        {isLoading && (
```

Comment:
Not sure why we would wish to display a Skeleton element here? I think we may want to show the `LoadingTable` instead when refetching, even when the current state is `No subscriptions` :thinking:

Reply:
The loading animation is only shown for a refetch, which means the user either sees the empty-state or the table before. 

If the user sees -> empty state -> loading table -> empty state I feel it would feel more confusing / disruptive then showing a loading version of the state they just saw.

Reply:
I would not show any loading state for the `EmptyState` component then. It renders no data, so no data is being refreshed :o

Reply:
I'm not sure if it's enough to show users that something is happening right now.

I didn't change the empty state in the beginning, but then it was really hard to see that something is reloading in the background. I think it would be a good approach to say we show a loading info for all areas where the date can change. 

In this case it would make it clear to the user that they have to wait until the loading has ended before they know if something did change here. 

I know that the loading for empty state implementation looks kind of weird (and it is). I first thought about showing a generic Table Skeleton, but it diverges from the pattern we're following now everywhere else in that area, which is keeping the layout as much as possible and showing skeletons as loading animations where the content can change. 🤔  

Also I just saw that the loading does not look correct, because the headline should not be visible while loading 🙈  only the bordered box with two skeletons.

Reply:
> Also I just saw that the loading does not look correct, because the headline should not be visible while loading 🙈 only the bordered box with two skeletons.

Haha yes, that was also a little weird to me.

> I first thought about showing a generic Table Skeleton

I still think this is the more correct way to show the loading state: we are expecting a table, not showing a table is the exception